src/compiler/builder.ts

```diff
@@ -132,6 +157,38 @@ namespace ts {
         return state;
     }
 
+    /**
+     * Releases program and other related not needed properties
+     */
+    function releaseCache(state: BuilderProgramState) {
+        BuilderState.releaseCache(state);
+        state.program = undefined;
+    }
+
+    /**
+     * Creates a clone of the state
+     */
+    function cloneBuilderProgramState(state: Readonly<BuilderProgramState>): BuilderProgramState {
+        const newState = BuilderState.clone(state) as BuilderProgramState;

```

Comment:
This seems weird -- if BuilderState can clone itself, why do we have to copy a bunch of properties here too?

Reply:
Because you cloned BuilderState and this is BuilderProgramState (interface BuilderProgramState extends BuilderState)

Reply:
I mean these are explicit properties in `BuilderProgramState`